microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIConfigFileHealthTest.java

```diff
@@ -0,0 +1,206 @@
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+
+/**
+ * MP FT integration will be tested relying on microprofile-config.properties,
+ * MP Config properties are defined in microprofile-config.properties in META INF.
+ *
+ * Testclass to test {@code /health}, {@code /health/live}, {@code /health/ready} endpoints.
+ * {@link CDIBasedLivenessHealthCheck} and {@link CDIBasedReadinessHealthCheck} probes are based on fail-safe CDI bean
+ * {@link FailSafeDummyService} configured at the beginning via MP Config properties:
+ * <ul>
+ * <li>{@link FailSafeDummyService#LIVE_CONFIG_PROPERTY} - service is live</li>
+ * <li>{@link FailSafeDummyService#READY_CONFIG_PROPERTY} - service is ready if
+ * {@link FailSafeDummyService#IN_MAINTENANCE_CONFIG_PROPERTY} is {@code false}</li>
+ * <li>{@link FailSafeDummyService#READY_IN_MAINTENANCE_CONFIG_PROPERTY} - service is ready if
+ * {@link FailSafeDummyService#IN_MAINTENANCE_CONFIG_PROPERTY} is {@code true}</li>
+ * </ul>
+ * If service is in maintenance a method simulating opening resources {@link FailSafeDummyService#simulateOpeningResources}
+ * throws a {@link IOException}.
+ *
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(MicroProfileFTSetupTask.class)
+public class FailSafeCDIConfigFileHealthTest {
+
+    private static final String MPConfigContent = String.format("%s=%s\n%s=%s\n%s=%s\n%s=%s",
+            FailSafeDummyService.LIVE_CONFIG_PROPERTY, false,
+            FailSafeDummyService.READY_CONFIG_PROPERTY, true,
+            FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY, true,
+            FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY, false);
+
+    protected RequestSpecification metricsRequest;
+
+    @ArquillianResource
+    protected ManagementClient managementClient;
+
+    @Before
+    public void before() throws ConfigurationException, InterruptedException, TimeoutException, IOException {
+        ArquillianContainerProperties arqProps = new ArquillianContainerProperties(
+                ArquillianDescriptorWrapper.getArquillianDescriptor());
+        String url = "http://" + arqProps.getDefaultManagementAddress() + ":" + arqProps.getDefaultManagementPort()
+                + "/metrics";
+        metricsRequest = given().baseUri(url).accept(ContentType.JSON);
```

Comment:
I was wondering whether it is mandatory that what is above - i.e. `metricsRequest` configuration is done before _each_ test. Wouldn't `@BeforeClass` be enough? Also with some concern about the order of extended `@Before` methods execution.

Reply:
The class no longer extends anything. I will move it to `@BeforeClass`

Reply:
So why is `reload` needed _before each method execution, now? Just asking, I am aware I might be missing something.

Reply:
Some tests check MP Metrics. Without reload the metric from a test would be preserved and have an impact on other tests.

Reply:
Thanks, see my comment below.